news: add 1.0.1 entry (#25)
groonga/groonga-nginx: NEWS.md

# NEWS

## 1.0.1 - 2025-02-21

### Package distribution update

In previous releases, Ubuntu packages were available only through the Groonga
APT repository. Starting with this release, packages are also distributed via

> packages.groonga.org.

Reviewer: What is the difference between "the Groonga APT repository" and "package.groonga.org"?

Author: fix: [9dd1041](https://github.com/groonga/groonga-nginx/pull/25/commits/9dd1041c02677f2452429d89a428b463eda7eb3d) I added the difference about the dependency package.

Reviewer: Does "the Groonga APT repository" mean https://launchpad.net/~groonga/+archive/ubuntu/ppa ?
We should call it as "the Groonga APT repository". We should call https://packages.groonga.org/ as "the Groonga APT repository" because packages.groonga.org is managed by us.

Author: I see. So both of them are "the Groonga APT repository".
I tried to rewrite it using specified url. What do you think of it? Or should I call the other ways?

Reviewer: Ah, sorry.

> We should call it as "the Groonga APT repository". 

↓

We should NOT call it as "the Groonga APT repository".

### Migration Notice for groonga-nginx

If you’re currently using groonga-nginx, we recommend migrating to the packages
provided by groonga.packages.org. Although the Groonga APT repository will still
be available, it will no longer receive new updates in near future.

For migration, please follow these steps:

```console
$ sudo add-apt-repository --remove ppa:groonga/ppa
$ sudo apt install -y -V ca-certificates lsb-release wget
$ wget https://apache.jfrog.io/artifactory/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
$ sudo apt install -y -V ./apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
$ rm -f apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb

> $ wget https://packages.groonga.org/debian/groonga-apt-source-latest-$(lsb_release --codename --short).deb

Reviewer: ```suggestion
$ wget https://packages.groonga.org/$(lsb_release --id --short | tr 'A-Z' 'a-z')/groonga-apt-source-latest-$(lsb_release --codename --short).deb
```

Author: fix: [aed39f2](https://github.com/groonga/groonga-nginx/pull/25/commits/aed39f2057fefca2f3420541cee110e9adfd3b7d)

$ sudo apt install -y -V ./groonga-apt-source-latest-$(lsb_release --codename --short).deb
$ rm -f groonga-apt-source-latest-$(lsb_release --codename --short).deb
$ sudo apt update

> $ sudo apt update -y -V libnginx-mod-http-groonga

Reviewer: Is "udpate" correct here?

Author: fix: [71988b1](https://github.com/groonga/groonga-nginx/pull/25/commits/71988b135d523d15c94dd76efe068fc61d079854) It's not correct. I fixed it using `install`.

> $ sudo apt update -y -V libnginx-mod-http-groonga

Reviewer 2: ```suggestion
$ sudo apt install -y -V libnginx-mod-http-groonga
```

Author: fix: [71988b1](https://github.com/groonga/groonga-nginx/pull/25/commits/71988b135d523d15c94dd76efe068fc61d079854) Thanks you so much!

# NEWS

## 1.0.1 - 2025-02-21

### Package distribution update

In previous releases, Ubuntu packages were available only through

> https://launchpad.net/~groonga/+archive/ubuntu/ppa. Starting with this release,

Reviewer: https://groonga.org/docs/install/ubuntu.html#ppa-personal-package-archive said "PPA on Launchpad". In general, users don't see https://launchpad.net/~groonga/+archive/ubuntu/ppa URL when they use. Because they use `add-apt-repository ppa:groonga/ppa`.

BTW, it said that "The Groonga APT repository for Ubuntu"...

packages.groonga.org.

While the groonga-nginx package itself remains unchanged, the underlying Groonga
dependency is different. The packages from packages.groonga.org include Groonga

> built with Apache Arrow enabled. Which unlocks extra features such as static

Reviewer: ```suggestion
built with Apache Arrow enabled. Which unlocks extra features such as offline
```

https://groonga.org/docs/reference/indexing.html#offline-index-construction

Author: fix: [ad73c68](https://github.com/groonga/groonga-nginx/pull/25/commits/ad73c68eab971caee09ece0c2993d5c4ec5be84a)

parallel index building.

> ### Migration Notice for groonga-nginx

Reviewer 2: Migration should only be required for Ubuntu users, so it is better to add that.

Author: fix: [dbc6986](https://github.com/groonga/groonga-nginx/pull/25/commits/dbc6986f5c3faade4d9bd7080d92e669fbeab948)

> If you’re currently using groonga-nginx, we recommend migrating to the packages

Reviewer: ```suggestion
If you're currently using groonga-nginx, we recommend migrating to the packages
```

Author: fix: [15840ab](https://github.com/groonga/groonga-nginx/pull/25/commits/15840ab6c91bf4d8a8f604dceb616b9dc585763a)

provided by groonga.packages.org. Although the Groonga APT repository will still
be available, it will no longer receive new updates in near future.

For migration, please follow these steps:

```console
$ sudo add-apt-repository --remove ppa:groonga/ppa
$ sudo apt install -y -V ca-certificates lsb-release wget
$ wget https://apache.jfrog.io/artifactory/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
$ sudo apt install -y -V ./apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
$ rm -f apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
$ wget https://packages.groonga.org/$(lsb_release --id --short | tr 'A-Z' 'a-z')/groonga-apt-source-latest-$(lsb_release --codename --short).deb
$ sudo apt install -y -V ./groonga-apt-source-latest-$(lsb_release --codename --short).deb
$ rm -f groonga-apt-source-latest-$(lsb_release --codename --short).deb
$ sudo apt update

> $ sudo apt install -y -V libnginx-mod-http-groonga

Reviewer 2: ppaからインストールした `libgroonga0` とかが残っていても大丈夫ですか？
`libnginx-mod-http-groonga` はまだ packages.groonga.org にないですが、これで移行できるかどのように確認しました？

Author: 現状確認できないので、リリースしてから確認するしかないのかなぁと考えていました。
ここには詳細に書かないで、README.mdを見てねと促して、あとから確認できた手段を明記したほうが良さそうですかねmm

Reviewer 2: 不確定な情報を記載するよりは、確実な情報を記載したほうが良いので、移行方法はあとからREADMEとかに書くほうが良かろうと思います。
（MIGRATION.md とか別ファイルを作っても良さそうです。）

Reviewer: Groongaの方に書いてそっちを参照でもいいかも。

Groongaの方はもうリリース済みだから動作確認できるよね？

Author: 一旦、ここまでで作業を止めてGroonga側に移行方法のドキュメントを書こうと思います。
